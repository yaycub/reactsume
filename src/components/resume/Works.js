import React from 'react';
import PropTypes from 'prop-types';
import WorkItem from './WorkItem';

const Works = ({ work }) => {
  const workElements = work.map((item, i) => {
    return (
      <li key={i}>
        <WorkItem {...item} />
      </li>
    );
  });

  return (
    <ul>
      {workElements}
    </ul>
  );
};

Works.propTypes = {
  work: PropTypes.array.isRequired
};

export default Works;
