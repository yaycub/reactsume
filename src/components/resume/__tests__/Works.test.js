import React from 'react';
import { shallow } from 'enzyme';
import data from '../../../resume.json';
import Works from '../Works.js';

describe('Works Component', () => {
  it('can render Works', () => {
    const wrapper = shallow(<Works {...data} />);
    expect(wrapper).toMatchSnapshot();
  });
});
