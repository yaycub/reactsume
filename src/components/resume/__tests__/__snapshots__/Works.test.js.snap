// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Works Component can render Works 1`] = `
<ul
  className="Works"
>
  <h3>
    Professional History
  </h3>
  <li
    key="0"
  >
    <WorkItem
      business="Airbnb"
      didWhat={
        Array [
          "Identified impact of bugs using SQL queries and notified engineering via Jira",
          "Built a partnership between Airbnb Collections and Airbnb Payments CS",
          "Trained multiple entry level associates on the Payments CS team & partner CS",
          "Consistent top performer on the team",
        ]
      }
      titleWithDates="Senior Payments Specialist | May 2014 - April 2019"
    />
  </li>
  <li
    key="1"
  >
    <WorkItem
      business="Capital One"
      didWhat={
        Array [
          "Managed a customer base to help them be up-to-date on credit payments",
          "QA for new hires and provided coaching on soft skills",
          "Provided and visualized data on successful attempts on collecting debt to help new hires with prioritizing workload",
        ]
      }
      titleWithDates="Risk Associate | September 2009 - December 2013"
    />
  </li>
</ul>
`;
